cli/tests/test_help_command.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2024 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::common::TestEnvironment;
+
+#[test]
+fn test_help() {
```

Comment:
nit: some other edge cases that might be interesting:

* `["help", "help"]`
* `["help", "unknown"]`

Maybe we can also insert a workaround for `["help", "log", "--", "-r"]`, but I personally don't care.

Reply:
maybe rem

> Maybe we can also insert a workaround for `["help", "log", "--", "-r"]`, but I personally don't care.

Maybe removing all the args that are not GlobalArgs? But, meh, is it worth it?

Reply:
> Maybe we can also insert a workaround for `["help", "log", "--", "-r"]`, but I personally don't care.

Alternatively, we can do subcommand lookup (as you did in earlier version), and print help message there instead of manipulating command arguments. It will require a bit more code, but I think it's less hacky.
https://docs.rs/clap/latest/clap/struct.Command.html#method.render_long_help

Since `jj help` is our own help command, we wouldn't have to copy the exact behavior on `jj help unknown subcommand`.

Reply:
But the previous subcommand lookup would break if you put some flags in the middle, as it would treat the flag as a subcommand.

Reply:
Just to be sure, with the current implementation, `help log -- -r` will be transformed to `log -r --help`. The error message is a bit cryptic. Instead, `help log -- -r` should ideally error out saying that the subcommand "log -r" doesn't exist.

Reply:
I agree with Yuya, but I also think the current state of the PR already seems like an improvement, so I'm personally fine with merging it and improving it in a follow-up if you prefer, Grillo-0. But let's also hear what Yuya thinks.

Reply:
A test for `help log -- r`?

Reply:
Yes, `help log -- -r` or something that produces a weird command arguments.